electron/__tests__/runtimes/utils.test.ts

```diff
@@ -0,0 +1,254 @@
+
+  it('Empty data', async () => {
+    expect.assertions(1)
+
+    jest.spyOn(axios, 'get').mockResolvedValue({ data: {}, status: 200 })
+    await expect(getAssetDataFromDownload(testUrl)).rejects.toEqual(
+      Error('Asset metadata could not be found')
+    )
+  })
+
+  it('HTTP error code', async () => {
+    expect.assertions(1)
+
+    jest.spyOn(axios, 'get').mockResolvedValue({ data: {}, status: 404 })
+    await expect(getAssetDataFromDownload(testUrl)).rejects.toEqual(
+      Error('Got HTTP error code 404')
+    )
+  })
+
+  // https://axios-http.com/docs/handling_errors
+  it('Axios error', async () => {
+    expect.assertions(1)
+
+    jest.spyOn(axios, 'get').mockRejectedValue({
+      toJSON: () => '{ "message": "Some error message" }'
+    })
+    await expect(getAssetDataFromDownload(testUrl)).rejects.toEqual(
+      Error('Failed to access GitHub API: { "message": "Some error message" }')
+    )
+  })
+})
+
+describe('downloadFile', () => {
+  it('Success', async () => {
+    const expectedData = readFileSync(testTarFilePath)
+
+    jest.spyOn(axios, 'get').mockResolvedValue({
+      status: 200,
+      data: expectedData
+    })
+    jest
+      .spyOn(graceful_fs, 'writeFile')
+      .mockImplementation(
+        (
+          path: any,
+          data: any,
+          callback: (err: NodeJS.ErrnoException | null) => void
+        ) => {
+          callback(null)
+        }
+      )
+
+    const tmpFileName = '/tmp/someFile'
+    await expect(downloadFile(testUrl, tmpFileName)).resolves.toBeUndefined()
+    expect(graceful_fs.writeFile).toBeCalledWith(
+      tmpFileName,
+      expectedData,
+      expect.anything()
+    )
+  })
+
+  it('Axios error', async () => {
+    expect.assertions(1)
+
+    jest.spyOn(axios, 'get').mockRejectedValue({
+      toJSON: () => '{ "message": "Some error message" }'
+    })
+
+    await expect(downloadFile(testUrl, '')).rejects.toEqual(
+      Error(
+        `Failed to download ${testUrl}: { "message": "Some error message" }`
+      )
+    )
+  })
+
+  it('HTTP error', async () => {
+    expect.assertions(1)
+
+    jest.spyOn(axios, 'get').mockResolvedValue({
+      status: 404,
+      data: {}
+    })
+    await expect(downloadFile(testUrl, '')).rejects.toEqual(
+      Error(`Failed to download ${testUrl}: HTTP error code 404`)
+    )
+  })
+
+  it('writeFile error', async () => {
+    const expectedData = readFileSync(
+      join(__dirname, 'test_data/TestArchive.tar.xz')
+    )
+
+    jest.spyOn(axios, 'get').mockResolvedValue({
+      status: 200,
+      data: expectedData
+    })
+
+    jest.spyOn(graceful_fs, 'writeFile').mockImplementation((fn, data, cb) => {
+      cb({ stack: 'Mocked error stack' } as Error)
+    })
+
+    await expect(downloadFile(testUrl, '')).rejects.toEqual(
+      Error('Failed to save downloaded data to file: Mocked error stack')
+    )
+  })
+})
+
+describe('extractTarFile', () => {
+  it('Success without strip', async () => {
+    const tmpDir = dirSync()
```

Comment:
This variable needs to be true, else tmp can't remove none empty folders.
```suggestion
    const tmpDir = dirSync({unsafeCleanup: true})
```

Doc:
```typescript
    /**
     * Recursively removes the created temporary directory, even when it's not empty.
     * @default false
     */
    unsafeCleanup?: boolean | undefined;
```